src/Models/BaseElement.php

```diff
@@ -528,6 +536,85 @@ public function getContentForSearchIndex(): string
         return $content;
     }
 
+    /**
+     * Provides content for CMS search
+     */
+    public function getContentForCmsSearch(): string
+    {
+        $textualDatabaseFields = $this->getTextualDatabaseFields();
+        $excludedFields = $this->getFieldsExcludedFromCmsSearch();
+        $contents = [];
+        foreach ($this->config()->get('db') as $fieldName => $fieldType) {
+            if (!in_array($fieldType, $textualDatabaseFields) || in_array($fieldName, $excludedFields)) {
+                continue;
+            }
```

Comment:
```suggestion
        $textualDatabaseFields = $this->getTextualDatabaseFields();
        $excludedFields = $this->getFieldsExcludedFromCmsSearch();
        $contents = [];
        foreach ($textualDatabaseFields as $fieldName) {
            if (in_array($fieldName, $excludedFields)) {
                continue;
            }
```
We already get the relevant field names in `getTextualDatabaseFields()` so there's no need to go through the db config a second time, unless I'm missing something?

Note depending on https://github.com/silverstripe/silverstripe-elemental/pull/1069#discussion_r1251324764 this may need an `array_keys()` call

Reply:
Seems correct, updated